moveit_planners/stomp/include/stomp_moveit/cost_functions.hpp

```diff
@@ -219,7 +220,28 @@ CostFn get_constraints_cost_function(const std::shared_ptr<const planning_scene:
 }
 
 /**
- * Creates a cost function that computes the summed waypoint penalites over a vector of cost functions.
+ * Creates a STOMP cost function from
+ *
+ * @param state_cost_function The planning scene instance to use for computing transforms
+ *
+ * @return                    STOMP Cost function based on MoveIt state cost function
+ */
+CostFn get_cost_function_from_moveit_state_cost_fn(const ::planning_interface::StateCostFn& state_cost_function)
+{
+  CostFn cost_fn = [&](const Eigen::MatrixXd& values, Eigen::VectorXd& costs, bool& validity) {
+    validity = true;  // TODO(sjahr): Discuss if that is always the case
+    for (int timestep = 0; timestep < values.cols() - 1; ++timestep)
```

Comment:
Is the cost not required for the last waypoint?

Reply:
Oh, you're right!